Dockerfile-velero-restore-helper.ubi

```
@@ -1,10 +1,16 @@
-FROM quay.io/konveyor/builder:latest AS builder
+FROM quay.io/konveyor/builder:ubi9-latest AS builder
```

```suggestion
FROM --platform=$BUILDPLATFORM quay.io/konveyor/builder:ubi9-latest AS builder
```

Since we are optimizing, and golang can cross compile binary, and this is not the final stage, we can set BUILDPLATFORM to get more optimum performance.

[ref: cross-compilation](https://docs.docker.com/build/building/multi-platform/#cross-compilation)

---

but this would make only multi platform builds faster, not?

since multi platform builds are CI responsibility (and I do not know how PROW does it), I think we do not need this

---

right do not need. but doesn't hurt to add for local dev builds either when building for dev/customers who are on arch other than the dev machine.

---

im on arm, so this would speed up builds of amd64 images to use on amd64 cluster.

---

This would make single platform build faster if the OS building (ie. laptop) is different than intended image platform